config/routes.rb

```diff
@@ -461,8 +461,11 @@
     post '/applications/:application_id/confirm-enrolment' => 'decisions#confirm_enrolment'
 
     post '/test-data/regenerate' => 'test_data#regenerate'
-    post '/experimental/test-data/generate' => 'test_data#generate'
-    post '/experimental/test-data/clear' => 'test_data#clear!'
+    post '/test-data/generate' => 'test_data#generate'
+    post '/test-data/clear' => 'test_data#clear!'
+
+    post '/experimental/test-data/generate' => redirect('/test-data/generate', status: 308)
```

Comment:
This gets me a routing error:

<img width="1319" alt="Screenshot 2020-07-08 at 16 54 28" src="https://user-images.githubusercontent.com/642279/86941313-b610b180-c13b-11ea-8f83-6480a4677989.png">

Adding `api/v1` to the front makes it work. Interestingly the params aren't passed so I generated the default 100 applications. The route redirect doesn't seem to work like a normal HTTP redirect — it seems to substitute one route for another. Have we got to redirect from the controller?

Reply:
Ah OK this didn't work as expected, I tested this with `CURL -XPOST localhost:3000/api/v1/experimental/test-data/clear` and got a 'You are being redirected' HTML response. I wasn't sure this was a desireable outcome. I'll take another look.

Reply:
I don't think we should redirect POST requests to the new/non-experimental endpoints. First of all, a full redirect (making the call succeed) is not an option, secondly, the calling code may not be able to follow redirects (HTTP/REST libraries used for API calls do not necessarily follow redirects automatically).

I think we should just return an appropriate HTTP error code (so they know their call was unsuccessful) and a JSON response explaining that the endpoints have moved (so they know what is wrong). Bonus points for telling consumers where the new endpoints are in the JSON response (so they know how to fix).

Reply:
@mnacos I've attempted this here https://github.com/DFE-Digital/apply-for-teacher-training/pull/2454/commits/469eef70c90639b0bfb227665e785538154e5984 I'm not sure about `410` as a response but all the redirect responses expect that we modify the `Location` header which isn't really what we want to do (Duncan also mentioned there is a side effect in redirecting). Opinions on which status to return welcome.

Reply:
I think `410 Gone` is perfect for this.